stdlib/uint32.gr

```diff
@@ -0,0 +1,494 @@
+/**
+ * Utilities for working with the Uint32 type.
+ * @example include "uint32"
+ * 
+ * @since v0.6.0
+ */
+
+module Uint32
+
+include "runtime/unsafe/wasmi32"
+include "runtime/unsafe/wasmi64"
+include "runtime/unsafe/memory"
+include "runtime/exception"
+include "runtime/numbers"
+from Numbers use { reducedUnsignedInteger, coerceNumberToUnsignedWasmI32 }
+include "runtime/dataStructures"
+from DataStructures use { newUint32 }
+
+@unsafe
+let _VALUE_OFFSET = 4n
+
+/**
+ * @section Conversions: Functions for converting between Numbers and the Uint32 type.
+ */
+
+/**
+ * Converts a Number to a Uint32.
+ *
+ * @param number: The value to convert
+ * @returns The Number represented as a Uint32
+ * 
+ * @since v0.6.0
+ */
+@unsafe
+provide let fromNumber = (x: Number) => {
+  let x = WasmI32.fromGrain(x)
+  Memory.incRef(x)
```

Comment:
This incRef shouldn't be necessary—is there something special about `coerceNumberToUnsignedWasmI32`?

Reply:
It is done for Int32s as well https://github.com/grain-lang/grain/blob/main/stdlib/runtime/numbers.gr#L2268 so I figured it'd just be safe to also include it even though I wasn't quite sure why it existed for signed ints

Reply:
Ah okay, that's a little different. In that case, since we're passing a Grain value as a WasmI32, the compiler doesn't know that it needs to incRef the value, so we do it manually. So it's really _just_ because of the `fromGrain`. You can skip the `fromGrain` here and pass `x` directly and the compiler will inject the incRef call for you.

Reply:
Ah ok I see